packages/esbuild/test/index.test.js

```diff
@@ -140,36 +140,30 @@ test('@mdx-js/esbuild', async () => {
   await fs.writeFile(new URL('esbuild.mdx', import.meta.url), 'a')
 
   console.log('\nnote: the following error is expected!\n')
-  try {
-    await esbuild.build({
+  await assert.rejects(
+    esbuild.build({
       entryPoints: [fileURLToPath(new URL('esbuild.md', import.meta.url))],
       outfile: fileURLToPath(new URL('esbuild-md-as-mdx.js', import.meta.url)),
       plugins: [esbuildMdx({format: 'mdx'})]
-    })
-    assert.unreachable()
-  } catch (error) {
-    assert.match(
-      String(error),
-      /No loader is configured for "\.md" files/,
-      'should not handle `.md` files w/ `format: mdx`'
-    )
-  }
+    }),
+    new Error(
+      'Build failed with 1 error:\nerror: No loader is configured for ".md" files: test/esbuild.md'
+    ),
```

Comment:
I have bad experiences with providing strings. I think errors work the same. Different versions of Node throw different errors. Paths use different slashes across OSes.
Can you reuse the regexes that existed before?